r3bbase/pars/R3BMSOffsetFinder.cxx

```diff
@@ -0,0 +1,179 @@
+/******************************************************************************
+ *   Copyright (C) 2019 GSI Helmholtzzentrum für Schwerionenforschung GmbH    *
+ *   Copyright (C) 2019-2023 Members of R3B Collaboration                     *
+ *                                                                            *
+ *             This software is distributed under the terms of the            *
+ *                 GNU General Public Licence (GPL) version 3,                *
+ *                    copied verbatim in the file "LICENSE".                  *
+ *                                                                            *
+ * In applying this license GSI does not waive the privileges and immunities  *
+ * granted to it by virtue of its status as an Intergovernmental Organization *
+ * or submit itself to any jurisdiction.                                      *
+ ******************************************************************************/
+
+// -------------------------------------------------------------------------
+// -----                R3BMSOffsetFinder source file                  -----
+// -----             Created 29/05/18  by J.L. Rodriguez-Sanchez       -----
+// ----               Adapted 26/07/23 by L.M. Fonseca                 -----
+// -------------------------------------------------------------------------
+
+// ROOT headers
+#include "R3BSamplerMappedData.h"
+#include "TClonesArray.h"
+#include "TF1.h"
+#include "TH1F.h"
+#include "TMath.h"
+#include "TObjArray.h"
+#include "TRandom.h"
+#include "TVector3.h"
+#include <iostream>
+#include <stdlib.h>
+// Fair headers
+#include "FairLogger.h"
+#include "FairRootManager.h"
+#include "FairRunAna.h"
+#include "FairRuntimeDb.h"
+
+// R3B headers
+#include "R3BEventHeader.h"
+#include "R3BMSOffsetFinder.h"
+#include "R3BMSOffsetPar.h"
+
+using namespace std;
+
+// R3BMSOffsetFinder: Default Constructor --------------------------
+R3BMSOffsetFinder::R3BMSOffsetFinder()
+    : R3BMSOffsetFinder("R3B MS Offset Finder", 1)
+{
+}
+
+// R3BMSOffsetFinder: Standard Constructor --------------------------
+R3BMSOffsetFinder::R3BMSOffsetFinder(const TString& name, Int_t iVerbose)
+    : FairTask(name, iVerbose)
+    , fMSOffsetPar(NULL)
+    , fSamplerMapped(NULL)
+    , fSamplerMSMapped(NULL)
+    , fMinStadistics(1000)
+{
+}
+
+// R3BMSOffsetFinder: Destructor ----------------------------------------
+R3BMSOffsetFinder::~R3BMSOffsetFinder() { LOG(info) << "R3BMSOffsetFinder: Delete instance"; }
```

Comment:
Don't declare deconstructor unless you have to.
```suggestion
```

Reply:
What YanzhaoW said.

Reply:
If I do that (the suggestion) this happens during compilation...
/usr/bin/ld: ../../../lib/libR3BBase.so.18.8.0: undefined reference to `vtable for R3BMSOffsetFinder'
/usr/bin/ld: ../../../lib/libR3BBase.so.18.8.0: undefined reference to `typeinfo for R3BMSOffsetFinder'
collect2: error: ld returned 1 exit status
make[2]: *** [neuland/executables/templates/CMakeFiles/ex_digi.dir/build.make:124: bin/ex_digi] Error 1
make[1]: *** [CMakeFiles/Makefile2:3007: neuland/executables/templates/CMakeFiles/ex_digi.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 96%] Built target R3BAsySource
/usr/bin/ld: ../../lib/libR3BBase.so.18.8.0: undefined reference to `vtable for R3BMSOffsetFinder'
/usr/bin/ld: ../../lib/libR3BBase.so.18.8.0: undefined reference to `typeinfo for R3BMSOffsetFinder'
collect2: error: ld returned 1 exit status
make[2]: *** [neuland/executables/CMakeFiles/neulandAna.dir/build.make:123: bin/neulandAna] Error 1
make[1]: *** [CMakeFiles/Makefile2:2972: neuland/executables/CMakeFiles/neulandAna.dir/all] Error 2
/usr/bin/ld: ../../lib/libR3BBase.so.18.8.0: undefined reference to `vtable for R3BMSOffsetFinder'
/usr/bin/ld: ../../lib/libR3BBase.so.18.8.0: undefined reference to `typeinfo for R3BMSOffsetFinder'
collect2: error: ld returned 1 exit status
make[2]: *** [neuland/executables/CMakeFiles/neulandSim.dir/build.make:165: bin/neulandSim] Error 1
make[1]: *** [CMakeFiles/Makefile2:2938: neuland/executables/CMakeFiles/neulandSim.dir/all] Error 2
/usr/bin/ld: ../../lib/libR3BBase.so.18.8.0: undefined reference to `vtable for R3BMSOffsetFinder'
/usr/bin/ld: ../../lib/libR3BBase.so.18.8.0: undefined reference to `typeinfo for R3BMSOffsetFinder'
collect2: error: ld returned 1 exit status
make[2]: *** [neuland/test/CMakeFiles/NeulandUnitTests.dir/build.make:189: bin/NeulandUnitTests] Error 1
make[1]: *** [CMakeFiles/Makefile2:2878: neuland/test/CMakeFiles/NeulandUnitTests.dir/all] Error 2
[ 97%] Built target R3BEvtVis
[100%] Built target R3BAnalysis
make: *** [Makefile:146: all] Error 2

Reply:
Did you remove the destructor declaration in the class declaration?

Reply:
You are right. Done.